block-storage/src/main/scala/coop/rchain/blockstorage/dag/codecs.scala

```diff
@@ -25,6 +26,19 @@ object codecs {
     blockMetadata => ByteVector(blockMetadata.toByteString.toByteArray)
   )
 
+  val codecBlockMessage = variableSizeBytesLong(uint32, bytes).xmap[BlockMessage](
+    byteVector => bytesToBlockMessage(byteVector.toArray),
+    blockMessage => ByteVector(blockMessageToBytes(blockMessage))
+  )
+
+  val codecApprovedBlock = variableSizeBytesLong(uint32, bytes).xmap[ApprovedBlock](
+    byteVector =>
+      ApprovedBlock
+        .from(ApprovedBlockProto.parseFrom(byteVector.toArray))
+        .getOrElse(ApprovedBlock.empty),
```

Comment:
We don't want to silently swallow errors. You can use `exmap` which supports errors. Empty definitions fro BlockMessaage and ApprovedBlock are not needed.